app/gui2/src/components/GraphEditor.vue

```diff
@@ -61,6 +61,21 @@ function keyboardBusy() {
   return document.activeElement != document.body
 }
 
+interface Interaction {
+  cancel?(): void
+  handleExprClick?(expr: ExprId): void
+  handleNodeClick?(node: ExprId): void
```

Comment:
prefer lambda syntax (`() => void`), parameters in method syntax are unsafely bivariant (not that it really matters here)